### ISSUE TITLE
Internal functional interface in Kotlin format in an uncompilable way

### ISSUE THREAD

**Issue by yonatankarp** [diffplug/spotless#1865]
If you're just submitting a feature request or question, no need for the above.

When I'm using an internal functional interface, the formatting puts the `fun` keyword in front of the `internal` keyword causing the code to be un-compileable.

For example, the following class:

```kotlin
internal fun interface DragonSlayingStrategy {
    fun execute()
}
```

Will end up as follows after running spotless:

```kotlin
fun internal interface DragonSlayingStrategy {
    fun execute()
}
```

The above code using Spotless Gradle plugin (version `6.22.0`) with this configuration:

```gradle
spotless {
        kotlin {
            target(
                fileTree("${project.rootDir}/$name") {
                    include("**/*.kt")
                },
            )
            // see https://github.com/shyiko/ktlint#standard-rules
            trimTrailingWhitespace()
            ktlint("1.0.1").editorConfigOverride(
                mapOf(
                    "ktlint_standard_package-name" to "disabled",
                    "ktlint_standard_enum-entry-name-case" to "disabled",
                    "ktlint_standard_trailing-comma" to "disabled",
                    "ktlint_standard_annotation" to "disabled",
                    "ktlint_standard_no-unused-imports" to "disabled",
                    "ktlint_standard_trailing-comma-on-call-site" to "disabled",
                    "ktlint_standard_trailing-comma-on-declaration-site" to "disabled",
                ),
            )
            diktat("1.2.3").configFile("${rootProject.rootDir}/config/diktat/diktat-analysis.yml")
        }
        kotlinGradle {
            target(
                fileTree(".") {
                    include("**/*.gradle.kts")
                },
            )
            trimTrailingWhitespace()
            indentWithTabs(2)
            indentWithSpaces(4)
            endWithNewline()
            ktlint()
        }
    }
```

Here you can see the output log of the issue:

```shell
 @@ -3,6 +3,6 @@
/**
·*·Strategy·interface.
·*/
-internal·fun·interface·DragonSlayingStrategy·{
+fun·internal·interface·DragonSlayingStrategy·{
····fun·execute()
}
```

**Comment by Goooler** [comment]
I can't repro it with Spotless or Ktlint CLI.

Write `DragonSlayingStrategy.kt`
```kotlin
internal fun interface DragonSlayingStrategy {
  fun execute()
}

```
Then
```sh
ktlint -F DragonSlayingStrategy.kt
```

Nothing changed.

**Comment by yonatankarp** [comment]
I'll create a simple project that reproduce the problem and share it here when it's ready 👍

**Comment by yonatankarp** [comment]
Hi @Goooler, here's the repo reproducing the issue I'm facing: https://github.com/yonatankarp/spotless-bug-example

it's worth mentioning that while creating it I noticed the following:

 - When setting the project for a single module everything worked fine (as you've mentioned)
 - When setting it as a multi-module project, where Spotless is applied for each module via the `build.gradle.kts` in the root the problem re-appears in the example.

It it 100% possible that I'm doing something wrong here and would be happy to hear if this is the case! 🙏

**Comment by Goooler** [comment]
It's something related to 
```kotlin
diktat("1.2.3").configFile("${rootProject.rootDir}/config/diktat/diktat-analysis.yml")
```
You can try `./gradlew spotlessApply` after removing this like above.

**Comment by yonatankarp** [comment]
Mmmm interesting, I'll try to figure out what are the where's the collision and open an issue for the diktat library 👍

**Comment by yonatankarp** [comment]
Well, after carefully checking the custom config - even after disabling it completely and running it as follows the issue still exists:

```kotlin
diktat("1.2.3")
```

I'll open an issue for them 👍

**Comment by yonatankarp** [comment]
Closing the loop: https://github.com/saveourtool/diktat/issues/1775

I'll close the ticket here once the issue would be solved so there would be no duplication by anyone else 😄

**Comment by yonatankarp** [comment]
The issue can be resolved by using version `1.2.5` or above of Diktat. Please look at [this](https://github.com/yonatankarp/spotless-bug-example) example for more info.

Closing the issue as the problem was resolved 😄

**Comment by Goooler** [comment]
By the way, you can move to Diktat 2.0 after #1972 is finished.
